src/reporters/composite_reporter.js

```diff
@@ -52,5 +59,12 @@ export default class CompositeReporter {
         }
```

Comment:
this is incorrect. Each call to nested reporter `close()` is async, so the callback will be called almost immediately without waiting for the nested reporters to close. We need to implement a simple wait group, e.g. (assuming `count` is captured by reference):
```
let count = 0;
cb = function() {
  count++;
  if (count >= this._reporters.length) {
    if (callback) {
      callback(); // call the real callback
    }
  }
};
for (let i = 0; i < this._reporters.length; i++) {
             let reporter = this._reporters[i];
             reporter.close(cb);
}

```